build.rs

```diff
@@ -63,8 +58,56 @@ fn configured_by_vcpkg() -> bool {
     false
 }
 
+fn static_or_not() {
+    use std::ascii::AsciiExt;
+
+    // On Windows-MSVC, always link dynamically
+    if cfg!(all(windows, target_env="msvc")) {
+        println!("cargo:rustc-link-lib=dylib=libpq");
+        return;
+    }
+
+    // Link unconditionally statically
+    if env::var_os("PQ_LIB_STATIC").is_some() {
+        println!("cargo:rustc-link-lib=static=pq");
```

Comment:
I've noticed that I need to include the following when linking `diesel` statically:

```rust
[dependencies]
# This is needed to make sure that Cargo statically links against
# `libssl`. This should happen automatically, but it doesn't.
openssl-sys = "0.9"
```

Is there an underlying issue here with `ps-sys` or `openssl-sys` when statically linking that we should deal with? Or am I just messing up? I have a [very small test program here](https://github.com/emk/rust-musl-builder/tree/master/examples/using-diesel) that might help figure this out.